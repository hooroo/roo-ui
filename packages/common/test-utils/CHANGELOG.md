# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
