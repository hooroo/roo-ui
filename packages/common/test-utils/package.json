{
  "name": "@roo-ui/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
