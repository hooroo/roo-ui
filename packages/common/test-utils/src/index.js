export { default as shallowWithTheme } from './shallowWithTheme'; // eslint-disable-line import/prefer-default-export
