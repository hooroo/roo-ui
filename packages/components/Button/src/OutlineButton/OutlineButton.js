import { themeGet, buttonStyle } from 'styled-system';
import Button from '..';

const getBackground = props => (props.bg
  ? themeGet(`colors.${props.bg}`, props.bg)(props)
  : buttonStyle(props).backgroundColor);

export default Button.extend`
  background-color: transparent;
  border-color: ${getBackground};
  transition: none;

  &:not(:hover), &:disabled {
    color: ${getBackground};
  }

  &:hover:not(:disabled) {
    background-color: ${getBackground};
    border-color: ${getBackground};
  }
`;
