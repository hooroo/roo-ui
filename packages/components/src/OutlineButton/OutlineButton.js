import { color, buttonStyle } from 'styled-system';
import { Button } from '..';

const getBackground = props =>
  color(props).backgroundColor || buttonStyle(props).backgroundColor;

const OutlineButton = Button.extend`
  background-color: transparent;
  border-color: ${getBackground};
  transition: none;

  &:not(:hover), &:disabled {
    color: ${getBackground};
  }

  &:hover:not(:disabled) {
    background-color: ${getBackground};
    border-color: ${getBackground};
  }
`;

OutlineButton.displayName = 'OutlineButton';

export default OutlineButton;
