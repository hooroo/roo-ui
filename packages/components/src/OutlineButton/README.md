# OutlineButton

<!-- STORY -->

## Installation

```shell
$ yarn add @roo-ui/components
```

## Example

```jsx
import { OutlineButton } from '@roo-ui/components';

export default (
  <OutlineButton>Hello world</OutlineButton>
);
```

## Properties

| Name       | Description           | Type   | Default | Required? |
|:-----------|:----------------------|:-------|:--------|:----------|
| `children` | button content        | `node` | -       | ✔︎         |
| `primary`  | use the primary color | `bool` | false   | -         |
| `rounded`  | use a rounded border  | `bool` | false   | -         |
| `disabled` | disable the button    | `bool` | false   | -         |

## Customization

This component can be customized with [styled-system](https://github.com/jxnblk/styled-system) by passing props for [space](https://github.com/jxnblk/styled-system#space-responsive), [color](https://github.com/jxnblk/styled-system#color-responsive), or [box shadow](https://github.com/jxnblk/styled-system#misc).
