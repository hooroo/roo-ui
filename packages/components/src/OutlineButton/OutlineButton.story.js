import React from 'react';
import { storiesOf } from '@storybook/react';
import { withDocs } from 'storybook-readme';
import { boolean, select } from '@storybook/addon-knobs/react';

import OutlineButton from './OutlineButton';
import README from './README.md';

storiesOf('Components|OutlineButton', module)
  .addDecorator(withDocs(README))
  .add('default', () => (
    <OutlineButton
      variant={select('Variant', ['default', 'primary'], 'default')}
      rounded={boolean('Rounded', false)}
      disabled={boolean('Disabled', false)}
    >
      Hello world
    </OutlineButton>
  ));