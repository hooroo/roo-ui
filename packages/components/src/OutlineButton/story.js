import React from 'react';
import { storiesOf } from '@storybook/react';
import { withDocs } from 'storybook-readme';
import { text, boolean } from '@storybook/addon-knobs/react';

import OutlineButton from './OutlineButton';
import README from './README.md';

storiesOf('OutlineButton', module)
  .addDecorator(withDocs(README))
  .add('default', () => (
    <OutlineButton
      primary={boolean('Primary', false)}
      rounded={boolean('Rounded', false)}
      disabled={boolean('Disabled', false)}
    >
      {text('Children', 'Hello world')}
    </OutlineButton>
  ));
