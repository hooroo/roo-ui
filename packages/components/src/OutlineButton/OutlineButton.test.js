import React from 'react';
import { qantas as theme } from '@roo-ui/themes';
import { mountWithTheme } from '@roo-ui/test-utils';
import { axe } from 'jest-axe';

import OutlineButton from '.';

describe('<OutlineButton />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mountWithTheme(
      <OutlineButton>Click here</OutlineButton>,
      theme,
    );
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has no accessibility errors', async () => {
    expect(await axe(wrapper.html())).toHaveNoViolations();
  });

  describe('primary', () => {
    beforeEach(() => {
      wrapper = mountWithTheme(
        <OutlineButton primary>Hello world</OutlineButton>,
        theme,
      );
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('rounded', () => {
    beforeEach(() => {
      wrapper = mountWithTheme(
        <OutlineButton rounded>Hello world</OutlineButton>,
        theme,
      );
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
