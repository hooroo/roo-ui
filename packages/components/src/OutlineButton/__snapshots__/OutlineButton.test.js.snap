// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OutlineButton /> primary renders correctly 1`] = `undefined`;

exports[`<OutlineButton /> renders correctly 1`] = `
.c0 {
  display: inline-block;
  margin: 0;
  padding: 0.75rem 1.5rem;
  font-size: 1rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.125em;
  -moz-letter-spacing: 0.125em;
  -ms-letter-spacing: 0.125em;
  letter-spacing: 0.125em;
  line-height: 1.35;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  border: 2px solid;
  border-color: transparent;
  border-radius: 4px;
  outline: 0;
  -webkit-transition: background-color 200ms ease-in;
  transition: background-color 200ms ease-in;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  color: white;
  background-color: #323232;
  background-color: transparent;
  border-color: #323232;
  -webkit-transition: none;
  transition: none;
}

.c0:hover {
  background-color: #181818;
}

.c0:hover:disabled {
  background-color: #323232;
}

.c0:focus {
  box-shadow: 0 0 2px 2px #999999;
}

.c0:disabled {
  opacity: 0.7;
  cursor: not-allowed;
}

.c0:not(:hover) {
  color: #323232;
}

.c0:hover:not(:disabled) {
  background-color: #323232;
  border-color: #323232;
}

.c0:disabled {
  color: #999999;
  border-color: #999999;
}

<OutlineButton
  variant="default"
>
  <StyledComponent
    forwardedRef={null}
    variant="default"
  >
    <Clean.button
      className="c0"
      variant="default"
    >
      <Clean.div
        className="c0"
        is="button"
        variant="default"
      >
        <button
          className="c0"
          variant="default"
        >
          Click here
        </button>
      </Clean.div>
    </Clean.button>
  </StyledComponent>
</OutlineButton>
`;

exports[`<OutlineButton /> rounded renders correctly 1`] = `undefined`;
